### PR TITLE
Add heterogenous examples

### PR DESCRIPTION
**Pull request by paulf81**
**Complete this sentence**
THIS PULL REQUEST __IS NOT__ READY TO MERGE

This pull requests proposes moving heterogenous examples into subfolder and adds 3 small examples to test very basic implementation.  Also confirms grid points are only on turbines, even in heterogenous inflow cases

### CODE REVIEW

**Comment by rafmudaf** [comment]
@bayc I think you also had an idea for organizing the examples. Want to check this out?

File: examples/heterogenous_flow/example_h0002_het_show_gridpy [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2019 NREL
```
Comment (bayc) [inline examples/heterogenous_flow/example_h0002_het_show_gridpy]: This file needs a '.' for the extension.

File: examples/heterogenous_flow/example_h0002_het_show_gridpy [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import numpy as np
+
+fi = wfct.floris_interface.FlorisInterface("heterogeneous_input_small.json")
+fi.calculate_wake()
+
+# Introduce variation in wind speed
+fi.reinitialize_flow_field(wind_speed=[6,9])
+fi.calculate_wake()
+
+# Show the grid points (note only on turbines, not on wind measurements)
```
Comment (bayc) [inline examples/heterogenous_flow/example_h0002_het_show_gridpy]: Why is this only on turbines?

File: examples/heterogenous_flow/example_h0002_het_show_gridpy (hunk 1 above)
Comment (paulf81) [reply examples/heterogenous_flow/example_h0002_het_show_gridpy]: Fixed thanks!

**Pull request merged**